WalletLibrary/WalletLibrary/Requests/Handlers/OpenId4VCIHandler.swift

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+enum OpenId4VCIHandlerError: Error
+{
+    case Unimplemented
+    case InputNotSupported
+}
+
+/**
+ * Handles a Raw Request expected to be a Credential Offer and configures a VerifiedIdRequest object.
+ */
+struct OpenId4VCIHandler: RequestHandling
+{
+    private let configuration: LibraryConfiguration
+    
+    init(configuration: LibraryConfiguration) 
+    {
+        self.configuration = configuration
+    }
+    
+    /// Determines whether a given raw request can be handled by this handler.
+    ///
+    /// This method checks if the raw request can be cast to a dictionary and contains
+    /// a specific key (`credential_issuer`), indicating it is a valid request for processing.
```

Comment:
Are we using this field because it is a mandatory field required for the flow to continue to validate if it is a valid request? Could we use this method to perform validation of all fields needed to successfully complete the flow and fail close here before proceeding?

Reply:
Oh this is a good question, originally, this method is used to see if this handler can handle the raw payload. This is the minimum number of checks to see if this rawRequest is compatible with the Handler. We want to do the actual validation logic in the handle method. Let's actually talk offline about this because its an interesting thought.